Instagram.js

```diff
@@ -22,50 +22,46 @@ const patchPostMessageJsCode = `(${String(function () {
   window.postMessage = patchedPostMessage;
 })})();`;
 
-export default class Instagram extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      modalVisible: false,
-      key: 1,
-    };
-  }
+export default function Instagram(props) {
+  const { wrapperStyle, containerStyle, closeStyle, onLoginSuccess, onLoginFailure, redirectUrl, appId, appSecret, responseType, scopes, language = 'en', incognito = false } = props;
+  const [modalVisible, setModalVisible] = useState(false);
+  const [key, setKey] = useState(1);
+  const webViewRef = useRef(null);
 
-  show() {
-    this.setState({ modalVisible: true });
+  function show() {
+    setModalVisible(true);
   }
 
-  hide() {
-    this.setState({ modalVisible: false });
+  function hide() {
+    setModalVisible(false);
   }
 
-  async onNavigationStateChange(webViewState) {
+  async function onNavigationStateChange(webViewState) {
     const { url } = webViewState;
-    const { key } = this.state;
+
     if (
       webViewState.title === 'Instagram' &&
       webViewState.url === 'https://www.instagram.com/'
     ) {
-      this.setState({ key: key + 1 });
+      setKey((k) => k + 1);
     }
-    if (url && url.startsWith(this.props.redirectUrl)) {
-      this.webView.stopLoading();
+    if (url && url.startsWith(redirectUrl)) {
+      webViewRef.current.stopLoading();
       const match = url.match(/(#|\?)(.*)/);
       const results = qs.parse(match[2]);
-      this.hide();
+      hide();
       if (results.access_token) {
         // Keeping this to keep it backwards compatible, but also returning raw results to account for future changes.
-        this.props.onLoginSuccess(results.access_token, results);
+        onLoginSuccess(results.access_token, results);
```

Comment:
should we assume that `onLoginSuccess` is required every time?

Reply:
add a conditional to validate that `onLoginSuccess` exists

```suggestion
        if (onLoginSuccess) {
          onLoginSuccess(results.access_token, results);
        }
```